packages/cardano-services-client/src/TxSubmitProvider/TxSubmitApiProvider.ts

```diff
@@ -43,7 +43,11 @@ export class TxSubmitApiProvider implements TxSubmitProvider {
 
         if (typeof status === 'number' && status >= 400 && status < 500) this.#healthStatus = true;
 
-        throw new ProviderError(ProviderFailure.BadRequest, mapCardanoTxSubmitError(data), data as string);
+        throw new ProviderError(
+          ProviderFailure.BadRequest,
+          mapCardanoTxSubmitError(data),
+          typeof data === 'string' ? data : JSON.stringify(data)
```

Comment:
I think JSON.stringify will work for simple strings too:
```diff
-          typeof data === 'string' ? data : JSON.stringify(data)
+          typeof data === 'string' ? data : JSON.stringify(data)
```

Reply:
Yes it works, but it also encodes the string: it adds starting and ending double quotes and escapes newlines and double quotes.
I think it's better avoiding one more encoding level if not necessary.